### PR TITLE
Feature: filtering posting journal

### PR DESCRIPTION
**Pull request by jniles**
This feature implements a few filters for the Posting Journal.  See the screenshot below:

journalfilters
_Fig 1: Posting Journal Filter Form_

When the filters are applied, they show up in the filter bar, which is hidden whenever no filtering is applied.  See below

filterbar
_Fig 2: Posting Journal Filter Bar_

In order to make these filters powerful, some more infrastructure needs to be implemented.  See #876 for one idea.

At the moment, the only implemented filters are 
 1. transaction date
 2. user_id
 3. account_id
 4. description (fuzzy search)

... on the client, although the server supports filtering on every column of the Posting Journal.

These provide good examples of how to implement the filters, and any other developer can add in a filter or two.

---
Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

File: client/src/js/services/JournalService.js [hunk 1]
```diff
@@ -10,5 +10,49 @@ JournalService.$inject = ['PrototypeApiService'];
...
+    // returns columns from filters
+    return columns.filter(function (column) {
+      var value = params[column.field];
+
+      if (angular.isDefined(value)) {
+
+        // this is to temporarily reduce the size of the description field
+        // @TODO - find a better way of doing this
+        if (column.truncate) {
+          column.value = String(value).substring(0, column.truncate) + '... ';
```
Comment (sfount) [inline client/src/js/services/JournalService.js:45]: I don't know exactly how this truncated value is displayed but the field could be truncated by the view. 

HTML provides this exact behaviour with 

``` css
text-overflow : elipsis;
```

This assumes that each filter has a maximum width that they must respect. 

As this method seems to work I see no reason why we should favour one or the other, just an idea.

File: client/src/partials/journal/journal.html [hunk 2]
```diff
@@ -38,6 +38,24 @@
   </div>
 </div>
 
+<!-- @todo remove hardcoded styles -->
```
Comment (sfount) [inline client/src/partials/journal/journal.html:41]: 😆 👍

**Review by sfount — APPROVED** [review]
LGTM.

File: client/src/js/services/JournalService.js (hunk 1 above)
Comment (jniles) [reply client/src/js/services/JournalService.js:45]: Ah, if we put a maximum width on the `<label>`s, that would be a much better solution.  Didn't think about that. Let me see if it works!

**Comment by jniles** [comment]
@sfount, I propose that this issue should be worked on while working on #876.  Both will involve a refactor of the CSS anyway.  What do you think?

**Comment by sfount** [comment]
@jniles That sounds like a good idea - I will update the issue thread to research this idea.

**Pull request merged**